ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.tsx

```diff
@@ -1,78 +1,119 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
 import {
-  getMetaMaskAccounts,
-  getSelectedInternalAccount,
-} from '../../../selectors';
-import CustodyLabels from '../../../components/institutional/custody-labels/custody-labels';
-import PulseLoader from '../../../components/ui/pulse-loader';
-import { INSTITUTIONAL_FEATURES_DONE_ROUTE } from '../../../helpers/constants/routes';
-import { SWAPS_CHAINID_DEFAULT_BLOCK_EXPLORER_URL_MAP } from '../../../../shared/constants/swaps';
-import { CHAIN_IDS } from '../../../../shared/constants/network';
-import { shortenAddress } from '../../../helpers/utils/util';
-import Tooltip from '../../../components/ui/tooltip';
-import { useI18nContext } from '../../../hooks/useI18nContext';
-import {
-  mmiActionsFactory,
-  showInteractiveReplacementTokenBanner,
-} from '../../../store/institutional/institution-background';
-import {
-  Label,
-  Icon,
+  Box,
+  Button,
   ButtonLink,
+  ButtonSize,
+  ButtonVariant,
+  Icon,
   IconName,
   IconSize,
-  Box,
-  Button,
-  BUTTON_VARIANT,
-  BUTTON_SIZES,
+  Label,
   Text,
 } from '../../../components/component-library';
 import {
-  OverflowWrap,
-  TextColor,
-  JustifyContent,
   BlockSize,
   Display,
   FlexDirection,
   IconColor,
+  JustifyContent,
+  OverflowWrap,
+  TextColor,
 } from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
 import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+import { getMetaMaskAccounts } from '../../../selectors';
+import { getInstitutionalConnectRequests } from '../../../ducks/institutional/institutional';
+import { getSelectedInternalAccount } from '../../../selectors/selectors';
+import { toChecksumHexAddress } from '../../../../shared/modules/hexstring-utils';
+import { SWAPS_CHAINID_DEFAULT_BLOCK_EXPLORER_URL_MAP } from '../../../../shared/constants/swaps';
+import { CHAIN_IDS } from '../../../../shared/constants/network';
+import {
+  mmiActionsFactory,
+  showInteractiveReplacementTokenBanner,
+} from '../../../store/institutional/institution-background';
+import CustodyLabels from '../../../components/institutional/custody-labels';
+import { INSTITUTIONAL_FEATURES_DONE_ROUTE } from '../../../helpers/constants/routes';
+import PulseLoader from '../../../components/ui/pulse-loader';
+import Tooltip from '../../../components/ui/tooltip';
+import { getMostRecentOverviewPage } from '../../../ducks/history/history';
+import { shortenAddress } from '../../../helpers/utils/util';
 
-const getButtonLinkHref = ({ address }) => {
+const getButtonLinkHref = ({ address }: { address: string }) => {
   const url = SWAPS_CHAINID_DEFAULT_BLOCK_EXPLORER_URL_MAP[CHAIN_IDS.MAINNET];
   return `${url}address/${address}`;
 };
 
-export default function InteractiveReplacementTokenPage({ history }) {
+type LabelItem = {
```

Comment:
could we move the interface to a file that's more public accessible as institutional/custody/types;

Reply:
Yes, indeed, I'll address it in another PR

Reply:
https://consensyssoftware.atlassian.net/browse/MMI-5328